tests/components/sunweg/test_config_flow.py

```diff
@@ -0,0 +1,135 @@
+        )
+
+    assert result["type"] == "abort"
+    assert result["reason"] == "no_plants"
+
+
+async def test_multiple_plant_ids(hass: HomeAssistant) -> None:
+    """Test registering an integration and finishing flow with an entered plant_id."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    user_input = FIXTURE_USER_INPUT.copy()
+    plant_list = [deepcopy(SUNWEG_PLANT_RESPONSE), deepcopy(SUNWEG_PLANT_RESPONSE)]
+
+    with patch.object(
+        APIHelper, "authenticate", return_value=SUNWEG_LOGIN_RESPONSE
+    ), patch.object(APIHelper, "listPlants", return_value=plant_list):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input
+        )
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "plant"
+
+        user_input = {CONF_PLANT_ID: 123456}
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert result["data"][CONF_USERNAME] == FIXTURE_USER_INPUT[CONF_USERNAME]
+    assert result["data"][CONF_PASSWORD] == FIXTURE_USER_INPUT[CONF_PASSWORD]
+    assert result["data"][CONF_PLANT_ID] == 123456
+
+
+async def test_one_plant_on_account(hass: HomeAssistant) -> None:
+    """Test registering an integration and finishing flow with an entered plant_id."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    user_input = FIXTURE_USER_INPUT.copy()
+
+    with patch.object(
+        APIHelper, "authenticate", return_value=SUNWEG_LOGIN_RESPONSE
+    ), patch.object(
+        APIHelper,
+        "listPlants",
+        return_value=[deepcopy(SUNWEG_PLANT_RESPONSE)],
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input
+        )
+
+    assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert result["data"][CONF_USERNAME] == FIXTURE_USER_INPUT[CONF_USERNAME]
+    assert result["data"][CONF_PASSWORD] == FIXTURE_USER_INPUT[CONF_PASSWORD]
+    assert result["data"][CONF_PLANT_ID] == 123456
+
+
+async def test_existing_plant_configured(hass: HomeAssistant) -> None:
+    """Test entering an existing plant_id."""
+    entry = MockConfigEntry(domain=DOMAIN, unique_id=123456)
+    entry.add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    user_input = FIXTURE_USER_INPUT.copy()
+
+    with patch.object(
+        APIHelper, "authenticate", return_value=SUNWEG_LOGIN_RESPONSE
+    ), patch.object(
+        APIHelper,
+        "listPlants",
+        return_value=[deepcopy(SUNWEG_PLANT_RESPONSE)],
```

Comment:
same thing re: fixture for this one. You can also avoid the deepcopying by scoping the fixture down to the test